diff -up ecryptfs-utils-84/src/utils/mount.ecryptfs_private.c.fgetc ecryptfs-utils-84/src/utils/mount.ecryptfs_private.c
--- ecryptfs-utils-84/src/utils/mount.ecryptfs_private.c.fgetc	2011-01-11 10:44:20.000000000 +0100
+++ ecryptfs-utils-84/src/utils/mount.ecryptfs_private.c	2011-01-11 10:44:32.000000000 +0100
@@ -88,7 +88,8 @@ char *fetch_sig(char *pw_dir, int entry)
 /* Read ecryptfs signature from file and validate
  * Return signature as a string, or NULL on failure
  */
-	char *sig_file, c;
+	char *sig_file;
+	int c;
 	FILE *fh;
 	char *sig;
 	int i;
