diff -up ecryptfs-utils-84/src/key_mod/ecryptfs_key_mod_pkcs11_helper.c.fixsigness ecryptfs-utils-84/src/key_mod/ecryptfs_key_mod_pkcs11_helper.c
--- ecryptfs-utils-84/src/key_mod/ecryptfs_key_mod_pkcs11_helper.c.fixsigness	2011-01-04 14:15:58.711761181 +0100
+++ ecryptfs-utils-84/src/key_mod/ecryptfs_key_mod_pkcs11_helper.c	2011-01-04 14:15:58.744761153 +0100
@@ -1144,7 +1144,7 @@ static int tf_pkcs11h_key_x509file(struc
 			goto out;
 		}
 
-		if ((subgraph_key_ctx->pkcs11h_data.certificate_blob_size = i2d_X509 (x509, NULL)) < 0	) {
+		if ((ssize_t)(subgraph_key_ctx->pkcs11h_data.certificate_blob_size = i2d_X509 (x509, NULL)) < 0	) {
 			syslog(LOG_ERR, "PKCS#11: Cannot read decode certificate");
 			rc = -EIO;
 			goto out;
@@ -1165,7 +1165,7 @@ static int tf_pkcs11h_key_x509file(struc
 		 */
 		p = subgraph_key_ctx->pkcs11h_data.certificate_blob;
 
-		if ((subgraph_key_ctx->pkcs11h_data.certificate_blob_size = i2d_X509 (x509, &p)) < 0) {
+		if ((ssize_t)(subgraph_key_ctx->pkcs11h_data.certificate_blob_size = i2d_X509 (x509, &p)) < 0) {
 			syslog(LOG_ERR, "PKCS#11: Cannot read decode certificate");
 			goto out;
 		}
